sdk/metric/doc.go

```diff
@@ -39,6 +39,30 @@
 // Meter.RegisterCallback and Registration.Unregister to add and remove
 // callbacks without leaking memory.
 //
+// # Cardinality Limits
+//
+// Cardinality refers to the number of unique attribute combinations for a single
+// metric within one collection (export) interval. High cardinality can lead to
+// excessive memory usage, increased storage costs, and backend performance issues.
+//
+// The OpenTelemetry Go Metric SDK does not enforce a cardinality limit by default.
+// A limit can be configured via the API using the `WithCardinalityLimit`
+// option.
+//
+// When the configured limit is reached, any new attribute combination is not
+// tracked separately. Instead, its measurements are aggregated together under
+// a special attribute set containing `{"otel.metric.overflow": true}`.
+// This ensures total metric values (e.g., Sum, Count) remain correct for the
+// collection interval, but information about the specific dropped combinations
+// is not preserved.
+//
+// Recommendations:
+//
+//   - Set the limit based on the theoretical maximum combinations or expected
+//     active combinations per export interval.
+//   - Too high - wastes memory due to preallocation.
```

Comment:
Is this true?  Where do we preallocate up to the cardinality limit?

IMO, the main downside of a too high cardinality limit is that it increases the worst-case memory overhead of the SDK.

Reply:
That and it has downstream affects for databases that cannot handle the cardinality.